fs/mountfs.py

```diff
@@ -67,7 +69,7 @@ def __init__(self, auto_close=True):
         super(MountFS, self).__init__()
         self.auto_close = auto_close
         self.default_fs = MemoryFS()  # type: FS
-        self.mounts = []  # type: MutableSequence[Tuple[Text, FS]]
+        self.mounts = {}  # type: Dict[Text, FS]
```

Comment:
I think the reason a `list` was in use here was to preserve ordering of mounted files. Sure, a `dict` is fine for newer Python versions but I'd think this would change behaviour on Python 2.7. I think using `collections.OrderedDict` would work instead (and we can drop that when we move the code to PyFilesystem3).